enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+deciding if a network is able to reach other services in other networks.
+
+Furthermore, in the previous proposal it mentions using an annotation on the namespace to indicate if the network was
+configured by OVN-Kubernetes and is in use for the namespace. By utilizing a CRD, the status of the network CRD itself
+can be used to indicate whether it is configured by OVN-Kubernetes, eliminating the need for annotating a namespace. 
+
+### IP Addressing
+
+As previously mentioned, one of the goals is to allow user-defined networks to have overlapping pod IP addresses. This
+is enabled by allowing a user to configure what CIDR to use for pod addressing when they create the network. However,
+this range may not overlap with the default cluster CIDR used by the cluster default network today.
+
+Furthermore, the internal masquerade subnet and the Kubernetes service subnet will remain unique and will exist globally
+to serve all networks. The masquerade subnet must be large enough to accommodate enough networks. Therefore, the
+subnet size of the masquerade subnet is equal to the number of desired networks * 2, as we need 2 masquerade IPs per
+network. The masquerade subnet remains localized to each node, so each node can use the same IP addresses and the size
+of the subnet does not scale with number of nodes.
+
+The transit switch subnets may overlap between all networks. This network is just used for transport between nodes, and
+is never seen by the pods or external clients.
+
+The join subnet of the default cluster network may not overlap with the join subnet of user-defined networks. This is
+due to the fact that the pod is connected to the default network, as well as the user-defined primary network. The join
+subnet is SNAT'ed to from the GR of that network in order to facilitate ingress reply service traffic going back to the
+proper GR in case it traverses the overlay. For this reason, the pods may see this IP address and routes are added to the
+pod to steer the traffic to the right interface (100.64.0.0/16 is the default cluster network join subnet):
+
+```
+[root@pod3 /]# ip route show
+default via 10.244.1.1 dev eth0 
+10.96.0.0/16 via 10.244.1.1 dev eth0 
+10.244.0.0/16 via 10.244.1.1 dev eth0 
+10.244.1.0/24 dev eth0 proto kernel scope link src 10.244.1.8 
+100.64.0.0/16 via 10.244.1.1 dev eth0
+```
+
+Since the pod needs routes for each join subnet, any network that is attached to the pod needs a unique join subnet. For
+a pod connected to the default cluster network (100.64.0.0/16), a user-defined primary network (100.65.0.0/16), and a
+secondary network (100.66.0.0/16), the routing table would look like:
+
+```
+[root@pod3 /]# ip route show
+default via 10.245.1.1 dev eth1 
+10.96.0.0/16 via 10.245.1.1 dev eth1
+10.244.0.0/16 via 10.244.1.1 dev eth0
+10.245.0.0/16 via 10.245.1.1 dev eth1
+10.246.0.0/16 via 10.246.1.1 dev eth2
```

Comment:
We have to specify that this is for layer3, for layer2 we don't need route /16 route since all the nodes are at the same subnet.

Maybe you can use eth1 for layer3 network and eth2 for layer2 to see the difference

Reply:
good idea. I'll update that.